### PR TITLE
Ensure base Awake function is called in Cursor.cs

### PR DESCRIPTION
**Pull request by ei2kpi**
Update Cursor.cs so that awake function is public virtual
Update InteractiveMeshCursor.cs to call base.Awake otherwise base Cursor.cs awake function gets hidden and never gets called.

Base awake function has originalDefaultCursorDistance = DefaultCursorDistance; and since that never gets called, originalDefaultCursorDistance stays 0 and eventually DefaultCursorDistance becomes 0 as well causing an Error and the cursor to always be at position 0,0,0 when not gazing at anything.

### CODE REVIEW

**Comment by ei2kpi** [comment]
InteractiveMeshCursor prefab throws this error and this pull request fixes it:

transform.position assign attempt for 'InteractiveMeshCursor' is not valid. Input position is { NaN, NaN, NaN }.
UnityEngine.Transform:set_position(Vector3)
HoloToolkit.Unity.InputModule.Cursor:UpdateCursorTransform() (at Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs:376)
HoloToolkit.Unity.InputModule.InteractiveMeshCursor:UpdateCursorTransform() (at Assets/HoloToolkit/Input/Scripts/Cursor/InteractiveMeshCursor.cs:129)
HoloToolkit.Unity.InputModule.Cursor:Update() (at Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs:168)

**Comment by StephenHodgson** [comment]
Updated the target branch. Please resolve conflicts accordingly.

File: Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public bool IsVisible
 
         #region MonoBehaviour Functions
 
-        private void Awake()
+        public virtual void Awake()
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs]: should be protected virtual.

File: Assets/HoloToolkit/Input/Scripts/Cursor/InteractiveMeshCursor.cs [hunk 2]
```diff
@@ -51,8 +51,9 @@ public class InteractiveMeshCursor : Cursor
 
         private Vector3 mAwakeScale;
 
-        private void Awake()
+        public override void Awake()
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Cursor/InteractiveMeshCursor.cs]: also protected.

**Comment by StephenHodgson** [comment]
@ei2kpi if you don't mind, I've asked @shubhamarora28 to make the changes in his PR since he's already editing those file.

**Comment by ei2kpi** [comment]
Fine with me! just want to get it fixed :)

**Pull request closed without merging**